fs_overlay/opt/certs_manager/models/domain.rb

```diff
@@ -73,20 +73,50 @@ def name
     parsed_descriptor[:domain]
   end
 
+  def upstream_backend_name
+    "backend_" + parsed_descriptor[:domain]
+  end
+
+  def upstream_proto()
+    mode = parsed_descriptor[:mode]
+    return unless ["->", "=>"].include? mode
+
+    default = mode == "->" ? "http://" : "https://"
+    parsed_descriptor[:upstream_proto] || default
+  end
+
+  def upstreams
+    upstreams = parsed_descriptor[:upstreams].to_s.split("|").delete_if { |v| v.empty? }
+    upstreams.map do |v|
+      match = v.match /^(?<upstream>[^\[]+)(?:\[(?<parameters>.*)\])?$/
+      raise "Invalid upstream: #{v}" unless match
+
+      match.named_captures.transform_keys(&:to_sym)
+    end
+  end
+
+  def multiple_upstreams?
+    upstreams.length > 1
+  end
+
   def upstream
-    parsed_descriptor[:upstream] if parsed_descriptor[:mode] == '->'
+    return unless parsed_descriptor[:mode] == '->'
```

Comment:
I guess this is for backwards compatibility?

Reply:
Yes. It's important to return nil for static-pages and redirect-mode otherwise the condition in default.ssl.conf.erb would be broken. I will add a comment.